### DISCUSSION TITLE
myMPD as user?

### DISCUSSION THREAD

**Discussion by manuvazquez** [jcorporation/myMPD#1376] [Q&A] [answered]
Hi,

I'm trying to run myMPD as user with no luck. MPD is running fine after
```
systemctl --user start mpd
```
but when I run
```
systemctl --user start mympd
```
and then
```
systemctl --user status mympd
```
I get
```
systemd[760]: mympd.service: Scheduled restart job, restart counter is at 5.
systemd[760]: mympd.service: Start request repeated too quickly.
systemd[760]: mympd.service: Failed with result 'exit-code'.
systemd[760]: Failed to start myMPD server daemon.
```

Also, if I just run `mympd` from the command-line, I get
```
❯ mympd
12:08:21 ERROR    mympd           Cache dir: creating "/var/cache/mympd" failed
12:08:21 ERROR    mympd           File exists
Exiting erroneous, thank you for using myMPD
```

I don't know if that is relevant or no, but early on I was able to run myMPD as root (along with MPD also as root) without a glitch. Also, in the [documentation](https://jcorporation.github.io/myMPD/030-running/) it is mentioned that myMPD is meant to be run as user. So it's probably me doing something wrong.

Any clue?

**Comment by jcorporation** [comment]
The user unit starts myMPD with different config and cache dir and some environment variables set.

```
ExecStart=@CMAKE_INSTALL_FULL_BINDIR@/mympd -w %h/.config/mympd -a %h/.cache/mympd
Environment="MYMPD_HTTP_PORT=8080"
Environment="MYMPD_SSL_PORT=8443"
```

- `@CMAKE_INSTALL_FULL_BINDIR@`: is usually /usr/bin/mympd
- `%h`: is your home directory

Try to start myMPD as user with this arguments and evironment or/and try `journalctl -u mympd` to investigate the log.

**Comment by manuvazquez** [comment]
Thanks for the quick reply!!

`journalctl --user -u mympd` returns
```
systemd[760]: Started myMPD server daemon.
(mympd)[8742]: mympd.service: Changing to the requested working directory failed: No such file or directory
(mympd)[8742]: mympd.service: Failed at step CHDIR spawning /usr/bin/mympd: No such file or directory
systemd[760]: mympd.service: Main process exited, code=exited, status=200/CHDIR
systemd[760]: mympd.service: Failed with result 'exit-code'.
```

On the other hand,
```
❯ mympd -w ~/.config/mympd -a ~/.cache/mympd
08:17:42 NOTICE   mympd           Cache dir: "/home/manu/.cache/mympd"
08:17:42 NOTICE   mympd           Work dir: "/home/manu/.config/mympd"
08:17:42 NOTICE   mympd           Config dir: "/home/manu/.config/mympd/config"
08:17:42 NOTICE   mympd           Setting loglevel to NOTICE
08:17:42 NOTICE   mympd           Starting myMPD 18.2.1
ec78c  1 mongoose.c:7513:mg_open_listen bind: 13
ec78c  1 mongoose.c:4824:mg_listen      Failed: http://[::]:80, errno 13
08:17:42 EMERG    mympd           Can't bind to http://[::]:80
Exiting erroneous, thank you for using myMPD
```

Just in case, my *mpd.conf* is fairly minimal
```
music_directory		"/mnt/data/musica"

playlist_directory "~/.mpd/playlists"

db_file "~/.mpd/database"

state_file "~/.mpd/state"

sticker_file			"~/.mpd/sticker.sql"

metadata_to_use	"albumartist,artist,album,title,track,name,genre,date,composer,performer,disc"

auto_update	"yes"
input {
        plugin "curl"
}

audio_output {
	type		"pipewire"
	name		"PipeWire Sound Server"
}
```

**Comment by jcorporation** [comment, marked as answer]
> (mympd)[8742]: mympd.service: Failed at step CHDIR spawning /usr/bin/mympd: No such file or directory

Is there a `.config` directory in your home directory? If not create it.

> ec78c  1 mongoose.c:4824:mg_listen      Failed: http://[::]:80, errno 13

You must set the environment for the listening ports, a normal user process can not bind to ports lower than 1024.

Remove eventually created config: `rm -rf ~/.config/mympd ~/.cache/mympd`

Try again:

```sh
export MYMPD_HTTP_PORT=8080
export MYMPD_SSL_PORT=8443
mympd -w ~/.config/mympd -a ~/.cache/mympd
```

**Reply by manuvazquez** [reply]
That did the trick!! Thank you so much!! ...and nice software, by the way :)
